SmplEditor/MainWindow.xaml.cs

```diff
@@ -202,10 +215,41 @@ private void ImportPlaylist()
                 string safeName = safeNames[fileIdx];
                 string fileName = fileNames[fileIdx];
                 string extension = this.GetExtensionFromSafeName(safeName);
-                if (extension == ".xml"){
-                    // this file is iTunes file
+                if (extension == ".xml")
+                {
+                    // This file is iTunes file
+                    var itunes = new ITunesLibraryParser.ITunesLibrary(fileName);
+                    List<ITunesLibraryParser.Playlist> iPlaylists = itunes.Playlists.ToList();
+                    List<ITunesLibraryParser.Track> iTracks = itunes.Tracks.ToList();
+                    // Unlike smpl files, multiple playlists can come in at once.
+                    // And all duplicateless track library comes separate to the playlists.
+                    // Does iTunes library share the reference with the tracks in the playlist?
+                    // No
+
+                    // Registereing and linking can be done in the scale of the whole library
+                    // While registering, create a map of iTunes.Track --> registered_Song
+                    // There is a bit concern since the tracks from iTracks and iPlaylists
+                    //    only equals in value but not in references.
+
+                    // After registering, just go through the playlists
+                    // Per playlist, map the iTunes.Tracks --> Registered_Song
+                    List<Song> newSongs = new List<Song>();
+                    List<Song> existingSmplSongs = new List<Song>();
+                    var trackToSongLookup = this.linkTracksToLibrary(iTracks, this.songLibrary, newSongs, existingSmplSongs);
+                    this.unmatchediTunesTracks.AddRange(newSongs);
+                    System.Diagnostics.Debug.WriteLine("{1} - New tracks: {0}", newSongs.Count, "iTunes");
+                    foreach (var iPlaylist in iPlaylists)
```

Comment:
maybe it's better to change iPlaylist --> importingPlaylist, to match the SMPL version of the method.